### ISSUE TITLE
Require cycle warning

### ISSUE THREAD

**Issue by hardcodet** [typestack/class-transformer#358]
Probably minor, but a still a bit of a red flag: I'm getting the build warning about cyclic dependencies - maybe this can be optimized?

```
Require cycle: node_modules\class-transformer\TransformOperationExecutor.js -> node
_modules\class-transformer\storage.js -> node_modules\class-transformer\metadata\MetadataStorage.js -> node_modules\class-tr
ansformer\TransformOperationExecutor.js

Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
```

**Comment by allan-simon** [comment]
I'm having it to, @jotamorais  is it something that can easily be fixed ?

**Comment by caiobiodere** [comment]
Having same issue here any plans to fix this? Otherwise I will have to fork this project and fix it

**Comment by caiobiodere** [comment]
@jotamorais any plans on fixing this issue?

**Comment by jotamorais** [comment]
We are reviewing a few PRs that we need to merge and also preparing a release with a those PRs and then we will start tackling these issues.
If you have gone through the issue and have an idea on how to fix it, please, consider forking and submitting a PR and will gladly review and merge.

I appreciate your understanding and will keep you posted once the release is out.

**Comment by allan-simon** [comment]
I will try to propose a PR

**Comment by allan-simon** [comment]
https://github.com/typestack/class-transformer/pull/375/files should do the trick

**Comment by NoNameProvided** [comment]
Fixed in #380.
